### PR TITLE
refactor: Rename ForeignCoins into ForeignCoin

### PR DESCRIPTION
**Pull request by chixiaowen**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes: <PD-#1182>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: proto/fungible/genesis.proto [hunk 1]
```diff
@@ -11,6 +11,6 @@ option go_package = "github.com/zeta-chain/zetacore/x/fungible/types";
 // GenesisState defines the fungible module's genesis state.
 message GenesisState {
   Params params = 1 [(gogoproto.nullable) = false];
-  repeated ForeignCoins foreignCoinsList = 2 [(gogoproto.nullable) = false];
+  repeated ForeignCoin foreignCoinsList = 2 [(gogoproto.nullable) = false];
```
Comment (lumtis) [inline proto/fungible/genesis.proto:14]:
```suggestion
  repeated ForeignCoin foreignCoinList = 2 [(gogoproto.nullable) = false];
```

**Review by lumtis — COMMENTED** [review]
Looks good, we should also rename the `foreignCoinsList` field as well.

FYI: renamed `feat` -> `refactor`

File: proto/fungible/genesis.proto (hunk 1 above)
Comment (chixiaowen) [reply proto/fungible/genesis.proto:14]: ok, I will do it

**Comment by chixiaowen** [comment]
hello? @lumtis

**Comment by lumtis** [comment]
> hello? @lumtis

Hey @chixiaowen ,

As stated in https://github.com/zeta-chain/node#%EF%B8%8F-important-notice-code-freeze-for-security-audit-%EF%B8%8F we are currently performing a code freeze in the repo until 18 December.
This means we can keep this PR open for the time being and then merge it once the audit is completed and code freeze withdrawn.

**Review by lumtis — COMMENTED** [review]
Thanks for the contribution @chixiaowen, we can have a new look at the PR if you can address the conflicts.

**Comment by lumtis** [comment]
It looks inactive.
Closing it for now in case someone else want to have a chance to address the issue.
Feel free to reopen.

**Pull request closed without merging**